conda/common/iterators.py

```diff
@@ -18,3 +19,20 @@ def groupby_to_dict(keyfunc, sequence):
     for key, group in itertools.groupby(sequence, keyfunc):
         result[key].extend(group)
     return dict(result)
+
+
+T = TypeVar("T")
+
+
+def unique(sequence: Sequence[T]) -> Generator[T, None, None]:
```

Comment:
could be a `Sequence[Any]` or "hashable" (or just Sequence?) since we don't need every element in the sequence to be the same type

Reply:
oops, you're totally right

Reply:
actually try 'Iterable'

Reply:
This is okay but `Sequence` is about being able to index by integers, iterable is about for ... in